### ISSUE TITLE
[Bug] Multiclass Classification Produces Binary Labels

### ISSUE THREAD

**Issue by m-r-mccormick** [mdabros/XGBoostSharp#71]
Using a trivial test/train dataset for binary classification  (2 classes, 0 and 1 training labels), I am able to get 100% accuracy. However, when using a trivial test/train dataset with 3+ classes for multiclass classification (0, 1, 2, ... training labels), all returned predictions are `0` or `1` regardless of how many unique labels/classes were used for training, which degrades accuracy. I am also using [XGBoost4J](https://xgboost.readthedocs.io/en/latest/jvm/index.html) (also appears to be a wrapper around the native implementation) in Java the same way, and I had the same issue until I changed the parameter `eval_metric` from the default `logloss` to `mlogloss`. With `mlogloss` in Java, similar code returns more than 2 unique labels predictions. However, it looks like there is no `eval_metric` parameter in the `XGBClassifier` constructor or `ParameterNames` in this library. Since multiclass predictions are inaccurate, I marked this as a bug, but could this parameter be added?

```cs
XGBClassifier model;
        if (uniqueLabels.Count <= 2)
            // Returns 0 and 1 labels
            model = new XGBClassifier();
        else
            // Only returns 0 and 1 labels
            // Java: had to also set 'eval_metric' to 'mlogloss'
            //   rather than the default 'logloss' for it to work for multiclass
            model = new XGBClassifier(objective: "multi:softmax", numClass: uniqueLabels.Count);

model.Fit(trainData, trainLabels);

var predictions = model.Predict(testData);
```

**Comment by mdabros** [comment]
Hi @m-r-mccormick 

For multi-label, you have to use `objective: "multi:softmax", as you have already done, and then the `PredictRaw` method instead of `Predict`. The `Predict` method does the conversion from probabilities to labels - but only for the binary case:

```csharp
    public float[] Predict(DMatrix dMatrix)
    {
        var predictions = m_booster.Predict(dMatrix).Select(v => v > 0.5f ? 1f : 0f).ToArray();
        return predictions;
    }
```

This is quite confusion, and probably has to change. But for now you can use the `PredictRaw` method.

I will add a test to illustrate this:

```csharp
    [TestMethod]
    public void XGBClassifierTest_PredictRaw_Multi_Label()
    {
        float[][] trainData =
        [
            [ 1.0f, 2.0f ],
            [ 1.1f, 2.2f ],
            [ 2.0f, 1.0f ],
            [ 2.1f, 1.1f ],
            [ 3.0f, 3.0f ],
            [ 3.1f, 3.1f ],
        ];

        float[] labels = [0, 0, 1, 1, 2, 2];

        var sut = CreateSut(learningRate: 0.3f, objective: Objective.Multi.Softmax, numClass: 3);
        sut.Fit(trainData, labels);

        float[] expected = [0, 0, 0, 1, 2, 2];
        var actual = sut.PredictRaw(trainData);

        TestUtils.AssertAreEqual(expected, actual);
    }
```

With regards to the `eval_metric` it is not supported yet, but will be in the future.
